datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -242,16 +257,13 @@ impl CommonSubexprEliminate {
 
     /// Extracts common sub-expressions and rewrites `exprs_list`.
     ///
-    /// Returns a tuple of:
-    /// 1. The rewritten expressions
-    /// 2. An optional tuple that contains the extracted common sub-expressions and the
-    ///    original `exprs_list`.
+    /// Returns `FoundCommonExprs` recording the result of the extraction
     fn find_common_exprs(
         &self,
         exprs_list: Vec<Vec<Expr>>,
         config: &dyn OptimizerConfig,
         expr_mask: ExprMask,
-    ) -> Result<Transformed<(Vec<Vec<Expr>>, FindCommonExprResult)>> {
+    ) -> Result<Transformed<FoundCommonExprs>> {
```

Comment:
The point of the PR is to make this change and the rest of the changes are a consequence of doign so